limitador/src/storage/redis/redis_cached.rs

```diff
@@ -395,10 +388,108 @@ impl CachedRedisStorageBuilder {
     }
 }
 
+async fn update_counters<C: ConnectionLike>(
+    redis_conn: &mut C,
+    counters_and_deltas: HashMap<Counter, AtomicExpiringValue>,
+) -> Result<Vec<(Counter, i64, i64)>, StorageErr> {
+    let redis_script = redis::Script::new(BATCH_UPDATE_COUNTERS);
+    let mut script_invocation = redis_script.prepare_invoke();
+
+    let mut res: Vec<(Counter, i64, i64)> = Vec::new();
+    let now = SystemTime::now();
+    for (counter, delta) in counters_and_deltas {
+        let delta = delta.value_at(now);
+        if delta > 0 {
+            script_invocation.key(key_for_counter(&counter));
+            script_invocation.key(key_for_counters_of_limit(counter.limit()));
+            script_invocation.arg(counter.seconds());
+            script_invocation.arg(delta);
+            // We need to store the counter in the actual order we are sending it to the script
+            res.push((counter, 0, 0));
+        }
+    }
+
+    let span = trace_span!("datastore");
```

Comment:
I wonder if it would make sense to have this a `debug_span` to match the other `datastore` spans - I don't think it particularly matters though